src/carousel.js

```diff
@@ -31,21 +32,29 @@ class Carousel extends Component {
     this.setScrollHandler();
   }
 
+  Constants = {
+    dummyItem : '',
+  }
+
   initialize() {
     const {
+      alwaysCentralizeSelected,
       itemWidth,
-      separatorWidth,
       data,
       containerWidth,
-      initialIndex
+      initialIndex,
     } = this.props;
     this.currentIndex = initialIndex;
     this.scrollXBegin = 0;
     this.xOffset = new Animated.Value(0);
     this.halfContainerWidth = containerWidth / 2;
     this.halfItemWidth = itemWidth / 2;
+    this.alwaysCentralizeSelected = alwaysCentralizeSelected;
+    this.data = this.alwaysCentralizeSelected ?
```

Comment:
if `this.props.data` changes, `this.data` will not reflect that. It might be a problem but depends on carousel usage...

Reply:
I was about to suggest you to use `componentWillReceiveProps`. However it's now deprecated (https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) and you should use a memoization as described at https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization.

Reply:
I'm prioritizing fixing the scroll behavior, I will fix it after.